README.md

# tellme_alert
Customizable popup for flutter.

<p float="left">
<img src="https://raw.githubusercontent.com/TellMeCorp/tellme-alert/main/example/images/Screenshot_1617999487.png?token=AFPPHGPUY2FHB3Z4NED6RH3APHY5Y" width = 150>
<img src="https://raw.githubusercontent.com/TellMeCorp/tellme-alert/main/example/images/Screenshot_1617999541.png?token=AFPPHGONLNCRS6OU2ZB2RXLAPHZE4" width = 150>
<img src="https://raw.githubusercontent.com/TellMeCorp/tellme-alert/main/example/images/Screenshot_1617999546.png?token=AFPPHGOGCHBPEQ7IB6INY7DAPHZBY" width = 150>
<img src="https://raw.githubusercontent.com/TellMeCorp/tellme-alert/main/example/images/Screenshot_1617999550.png?token=AFPPHGPUZM7S7L5YJR5OPW3APHZGO" width = 150>
<img src="https://raw.githubusercontent.com/TellMeCorp/tellme-alert/main/example/images/Screenshot_1617999553.png?token=AFPPHGLTXJOXNB72MMA6NY3APHZTK" width = 150>
<img src="https://raw.githubusercontent.com/TellMeCorp/tellme-alert/main/example/images/Screenshot_1617999556.png?token=AFPPHGJAQ2QKBFOB23BXYIDAPHZNW" width = 150>
<img src="https://raw.githubusercontent.com/TellMeCorp/tellme-alert/main/example/images/Screenshot_1617999558.png?token=AFPPHGOKOMANKJAMQWATCB3APHZLC" width = 150>
</p>


## Install

### 1. Depend on it
Add this to your package's pubspec.yaml file:

```yaml
dependencies:
    tellme_alert:: <CurrentVersion>
```

### 2. Install it

You can install packages from the command line:  

with Flutter:
```shell
$ flutter pub get
```

Alternatively, your editor might support `flutter pub get`. Check the docs for your editor to learn more.

### 3. Import it

Now in your Dart code, you can use:

```dart
import  'package:tellme_alert/tellme_alert.dart';
```

## Example

  
```dart
//Default confirmed
TellMeAlert(
    context: context,
    title: "Lorem ipsum title",
    content:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
    onConfirm: () => print("Default confirmed"),
),
//Aligned
TellMeAlert(
    context: context,
    alignment: Alignment.topCenter,
    borderRadius: 10,
    iconColor: TellMeColors.red_500,
    confirmButtonColor: TellMeColors.red_500,
    title: "Lorem ipsum title",
    content:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
    onConfirm: () => print("Aligned confirmed"),
),
//Show false
TellMeAlert(
    context: context,
    alignment: Alignment.bottomCenter,
    borderRadius: 15,
    showCancelButton: false,
    showContent: false,
    showIcon: false,
    iconColor: TellMeColors.blue_500,
    confirmButtonColor: TellMeColors.blue_500,
    title: "Lorem ipsum title",
    onConfirm: () => print("Disabled confirmed"),
),
//Custom Widget
TellMeAlert(
    context: context,
    padding: const EdgeInsets.all(70),
    child: Container(
        color: Colors.amber,
        child: IconButton(
            icon: Icon(Icons.ac_unit),
            onPressed: () => Navigator.pop(context)),
    ),
    borderRadius: 30,
    showCancelButton: false,
    showContent: false,
    showConfirmButton: false,
    showTitle: false,
    showIcon: false,
    onConfirm: () => print("Custom widget confirmed"),
    ),
    child: Text("Custom widget element"),
),
```

CHANGELOG.md
## [0.0.2] - 13.04.2021

* Update readme.md

## [0.0.1] - 9.04.2021

* Tellme alert published as package.

test/tellme_alert_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:tellme_alert/tellme_alert.dart';

void main() {
  test('adds one to input values', () {
    final calculator = Calculator();
    expect(calculator.addOne(2), 3);
    expect(calculator.addOne(-7), -6);
    expect(calculator.addOne(0), 1);
  });
}


lib/tellme_alert.dart
import 'package:flutter/material.dart';

class TellMeAlert {
  final BuildContext context;
  //callback
  final Function onConfirm;
  //modal conf
  final Alignment alignment;
  final EdgeInsetsGeometry padding;
  final double borderRadius;
  final bool shadow;
  //Title conf
  final bool showTitle;
  final String title;
  final TextStyle titleStyle;
  //Content conf
  final bool showContent;
  final String content;
  final TextStyle contentStyle;
  //Icon conf
  final bool showIcon;
  final IconData icon;
  final Color iconColor;
  //Button conf
  final bool showConfirmButton;
  final String confirmButtonText;
  final TextStyle confirmTextStyle;
  final Color confirmButtonColor;
  final bool showCancelButton;
  final String cancelButtonText;
  final TextStyle cancelTextStyle;
  final Color cancelButtonColor;
  //custom content
  final Widget child;

  TellMeAlert({
    @required this.context,
    //callback
    this.onConfirm,
    //modal conf
    this.padding = const EdgeInsets.all(16.0),
    this.alignment = Alignment.center,
    this.borderRadius = 5,
    this.shadow = false,
    //title conf
    this.showTitle = true,
    this.title,
    this.titleStyle = _paragraphBold,
    //Content conf
    this.showContent = true,
    this.content,
    this.contentStyle = _uISubContentNormal,
    //icon conf
    this.showIcon = true,
    this.icon = Icons.error_outline,
    this.iconColor,
    //Button conf
    this.showConfirmButton = true,
    this.showCancelButton = true,
    this.confirmButtonText = "Confirm",
    this.cancelButtonText = "Cancel",
    this.confirmTextStyle,
    this.cancelTextStyle = _uISubContentNormal,
    this.confirmButtonColor,
    this.cancelButtonColor,
    //custom content
    this.child,
  }) {
    // assert(showTitle == true ? title != null : title == null,
    //     "If show title true, title can not be null");

    showAlert(context);
  }

  showAlert(BuildContext context) {
    Color primary = Theme.of(this.context).primaryColor;
    showDialog(
        barrierColor: Colors.black26,
        context: context,
        builder: (_) {
          return Align(
            alignment: this.alignment,
            child: Material(
              color: Colors.transparent,
              child: Padding(
                padding: this.padding,
                child: Container(
                  padding: const EdgeInsets.symmetric(vertical: 16),
                  decoration: BoxDecoration(
                    borderRadius: BorderRadius.circular(this.borderRadius),
                    boxShadow: [
                      this.shadow
                          ? BoxShadow(
                              color: Colors.black26,
                              blurRadius: 5,
                              spreadRadius: 1,
                            )
                          : BoxShadow()
                    ],
                    color: Colors.white,
                  ),
                  width: double.infinity,
                  child: SingleChildScrollView(
                    physics: BouncingScrollPhysics(),
                    child: Column(
                      children: [
                        this.showIcon
                            ? Padding(
                                padding: const EdgeInsets.all(16.0),
                                child: Container(
                                  decoration: BoxDecoration(
                                      shape: BoxShape.circle,
                                      color: this.iconColor != null
                                          ? this.iconColor.withOpacity(0.2)
                                          : primary.withOpacity(0.2)),
                                  child: Padding(
                                    padding: const EdgeInsets.all(16.0),
                                    child: Icon(
                                      this.icon,
                                      color: this.iconColor ?? primary,
                                      size: 45,
                                    ),
                                  ),
                                ),
                              )
                            : Container(),
                        showTitle
                            ? Padding(
                                padding: const EdgeInsets.symmetric(
                                    vertical: 8, horizontal: 16),
                                child: Text(
                                  this.title,
                                  textAlign: TextAlign.center,
                                  style: this.titleStyle,
                                ),
                              )
                            : Container(),
                        showContent
                            ? Padding(
                                padding: const EdgeInsets.symmetric(
                                    vertical: 8, horizontal: 16),
                                child: Text(
                                  this.content,
                                  textAlign: TextAlign.center,
                                  style: this.contentStyle,
                                ),
                              )
                            : Container(),
                        child == null ? Container() : child,
                        showConfirmButton
                            ? Container(
                                padding: const EdgeInsets.symmetric(
                                    vertical: 8, horizontal: 16),
                                width: double.infinity,
                                child: ElevatedButton(
                                  style: ButtonStyle(
                                    elevation: MaterialStateProperty.all(0),
                                    backgroundColor: MaterialStateProperty.all(
                                        this.confirmButtonColor ?? primary),
                                    shape: MaterialStateProperty.all<
                                        RoundedRectangleBorder>(
                                      RoundedRectangleBorder(
                                        borderRadius: BorderRadius.circular(
                                            this.borderRadius),
                                      ),
                                    ),
                                  ),
                                  onPressed: () {
                                    onConfirm();
                                    Navigator.pop(context);
                                  },
                                  child: Padding(
                                    padding: const EdgeInsets.all(16.0),
                                    child: Text(
                                      this.confirmButtonText,
                                      style: this.confirmTextStyle,
                                    ),
                                  ),
                                ),
                              )
                            : Container(),
                        showCancelButton
                            ? Container(
                                padding: const EdgeInsets.symmetric(
                                    vertical: 8, horizontal: 16),
                                width: double.infinity,
                                child: ElevatedButton(
                                  style: ButtonStyle(
                                    elevation: MaterialStateProperty.all(0),
                                    overlayColor: MaterialStateProperty.all(
                                        Colors.black12),
                                    backgroundColor: MaterialStateProperty.all(
                                        this.cancelButtonColor ?? Colors.white),
                                    shape: MaterialStateProperty.all<
                                        RoundedRectangleBorder>(
                                      RoundedRectangleBorder(
                                        borderRadius: BorderRadius.circular(
                                            this.borderRadius),
                                      ),
                                    ),
                                  ),
                                  onPressed: () => Navigator.pop(context),
                                  child: Padding(
                                    padding: const EdgeInsets.all(16.0),
                                    child: Text(
                                      this.cancelButtonText,
                                      style: this.cancelTextStyle,
                                    ),
                                  ),
                                ),
                              )
                            : Container(),
                      ],
                    ),
                  ),
                ),
              ),
            ),
          );
        });
  }
}

const _paragraphBold = const TextStyle(
  fontWeight: FontWeight.w700,
  color: const Color(0xff102d42),
  fontStyle: FontStyle.normal,
  fontSize: 16.0,
);

const _uISubContentNormal = const TextStyle(
  fontWeight: FontWeight.w500,
  color: const Color(0xff69819a),
  //fontFamily: "Gilroy-☞",
  fontStyle: FontStyle.normal,
  fontSize: 14.0,
);


example/tellme_alert_demo.dart
import 'package:flutter/material.dart';

class TellMeAlertPage extends StatefulWidget {
  @override
  _TellMeAlertPageState createState() => _TellMeAlertPageState();
}

class _TellMeAlertPageState extends State<TellMeAlertPage> {
  String log = "log";

  setLog(String message) {
    setState(() {
      log = message;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("TellMeAlert demo"),
      ),
      body: Container(
        width: double.infinity,
        child: Column(
          // mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.center,
          children: [
            TextButton(
              onPressed: () => TellMeAlert(
                context: context,
                title: "Lorem ipsum title",
                content:
                    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
                onConfirm: () => setLog("Default confirmed"),
              ),
              child: Text("Default alert"),
            ),
            TextButton(
              onPressed: () => TellMeAlert(
                context: context,
                iconColor: TellMeColors.green_500,
                confirmButtonColor: TellMeColors.green_500,
                cancelButtonColor: TellMeColors.green_300,
                cancelTextStyle: TextStyle(color: TellMeColors.white_100),
                title: "Lorem ipsum title",
                content:
                    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
                onConfirm: () => setLog("Custom color confirmed"),
              ),
              child: Text("Custom color"),
            ),
            TextButton(
              onPressed: () => TellMeAlert(
                context: context,
                borderRadius: 30,
                shadow: true,
                iconColor: TellMeColors.red_500,
                confirmButtonColor: TellMeColors.red_500,
                title: "Lorem ipsum title",
                content:
                    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
                onConfirm: () => setLog("Border radius confirmed"),
              ),
              child: Text("Border with shadow radius"),
            ),
            TextButton(
              onPressed: () => TellMeAlert(
                context: context,
                alignment: Alignment.topCenter,
                borderRadius: 10,
                iconColor: TellMeColors.red_500,
                confirmButtonColor: TellMeColors.red_500,
                title: "Lorem ipsum title",
                content:
                    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
                onConfirm: () => setLog("Aligned confirmed"),
              ),
              child: Text("Aligned"),
            ),
            TextButton(
              onPressed: () => TellMeAlert(
                context: context,
                alignment: Alignment.bottomCenter,
                borderRadius: 15,
                showCancelButton: false,
                showContent: false,
                showIcon: false,
                iconColor: TellMeColors.blue_500,
                confirmButtonColor: TellMeColors.blue_500,
                title: "Lorem ipsum title",
                onConfirm: () => setLog("Disabled confirmed"),
              ),
              child: Text("Disabled element"),
            ),
            TextButton(
              onPressed: () => TellMeAlert(
                context: context,
                alignment: Alignment.bottomCenter,
                borderRadius: 15,
                showContent: false,
                showIcon: false,
                confirmButtonColor: TellMeColors.orange_500,
                title: "Lorem ipsum title",
                titleStyle: TextStyle(color: TellMeColors.orange_500),
                cancelTextStyle: TextStyle(color: TellMeColors.orange_500),
                confirmTextStyle: TextStyle(color: TellMeColors.white_100),
                cancelButtonText: "İptal",
                confirmButtonText: "Onay",
                onConfirm: () => setLog("Custom text style confirmed"),
              ),
              child: Text("Custom text style element"),
            ),
            TextButton(
              onPressed: () => TellMeAlert(
                context: context,
                padding: const EdgeInsets.all(70),
                borderRadius: 30,
                showCancelButton: false,
                showContent: false,
                showConfirmButton: false,
                showTitle: false,
                icon: Icons.recommend,
                iconColor: TellMeColors.primary_500,
                onConfirm: () => setLog("Custom icon with padding confirmed"),
              ),
              child: Text("Custom icon with padding element"),
            ),
            TextButton(
              onPressed: () => TellMeAlert(
                context: context,
                padding: const EdgeInsets.all(70),
                child: Container(
                  color: Colors.amber,
                  child: IconButton(
                      icon: Icon(Icons.ac_unit),
                      onPressed: () => Navigator.pop(context)),
                ),
                borderRadius: 30,
                showCancelButton: false,
                showContent: false,
                showConfirmButton: false,
                showTitle: false,
                showIcon: false,
                onConfirm: () => setLog("Custom widget confirmed"),
              ),
              child: Text("Custom widget element"),
            ),
            Divider(
              thickness: 5,
            ),
            Text(log)
          ],
        ),
      ),
    );
  }
}
